hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1862,6 +1863,53 @@ private class Emit(
         }
         val moveArgs = refs.map( ref =>  Code(ref.m := ref.tempM, ref.v.storeAny(ref.tempV)) )
         EmitTriplet(Code(Code(storeTempArgs ++ moveArgs: _*), coerce[Unit](jump)), const(false), Code._empty)
+      case x@ReadValue(path, spec, requestedType) =>
+        val p = emit(path, env)
+        val pathString = path.pType2.asInstanceOf[PString].loadString(p.value[Long])
+        val rowBuf = spec.buildCodeInputBuffer(mb.fb.getUnsafeReader(pathString, true))
+
+        val (_, dec) = typeToTypeInfo(x.pType2) match {
```

Comment:
I'm not clear on the current state of affairs. Are we supposed to be using `ptype2` currently? I figured we'd just write `ptype` everywhere and at some point someone would do `def ptype = ptype2` or something.

Reply:
I'l change this to pType.